nav2_tasks/include/nav2_tasks/navigate_to_pose_task.hpp

```diff
@@ -39,22 +40,46 @@ class NavigateToPoseTaskServer : public TaskServer<NavigateToPoseCommand, Naviga
     goal_sub_ = node_->create_subscription<geometry_msgs::msg::PoseStamped>("move_base_simple/goal",
         std::bind(&NavigateToPoseTaskServer::onGoalPoseReceived, this, std::placeholders::_1));
 
+    initial_pose_received_ = false;
+    initial_pose_sub_ = node_->create_subscription<geometry_msgs::msg::PoseWithCovarianceStamped>(
+      "initialpose",
+      std::bind(&NavigateToPoseTaskServer::onInitialPoseReceived, this, std::placeholders::_1));
+
     // A client that we'll use to send a command message to our own task server
     self_client_ = std::make_unique<nav2_tasks::NavigateToPoseTaskClient>(node_);
   }
 
   NavigateToPoseTaskServer() = delete;
 
+  bool hasInitialPoseReceived()
+  {
+    if (initial_pose_received_) {
+      return true;
+    }
+    return false;
+  }
+
+  void setInitialPose(bool initial_pose)
```

Comment:
if `initial_pose_` is false, we don't need to pass an argument to make it true.

Reply:
In this case we do.  When AutoLocalization is done, this flags needs to be changed so that on next cycle we know that the robot has already been localized.